core/src/main/scripts/importer/validateData.py

```diff
@@ -2138,11 +2145,43 @@ class FusionValidator(Validator):
         'Method',
         'Frame']
     REQUIRE_COLUMN_ORDER = True
+    ALLOW_BLANKS = True
+
+    def __init__(self, *args, ** kwargs):
+        super(FusionValidator, self).__init__(*args, **kwargs)
+        self.fusion_lines = {}
 
     def checkLine(self, data):
+
         super(FusionValidator, self).checkLine(data)
-        # TODO check the values
 
+        # parse hugo and entrez to validate them together
+        hugo_symbol = None
+        entrez_id = None
+        if 'Hugo_Symbol' in self.cols:
+            hugo_symbol = data[self.cols.index('Hugo_Symbol')].strip()
+            # treat the empty string or 'Unknown' as a missing value
+            if hugo_symbol in ('', 'Unknown'):
+                hugo_symbol = None
+        if 'Entrez_Gene_Id' in self.cols:
+            entrez_id = data[self.cols.index('Entrez_Gene_Id')].strip()
+            # treat the empty string or 0 as a missing value
+            if entrez_id in ('', '0'):
+                entrez_id = None
+        # validate hugo and entrez together:
+        self.checkGeneIdentification(hugo_symbol, entrez_id)
+
+        # validate uniqueness of lines
+        fusion_line = "\t".join(data)
+        if fusion_line in self.fusion_lines:
```

Comment:
strictly speaking this is not what the loader does right? This is potentially less strict. Lines that are the same but have an extra optional column that is not parsed by loader will be seen as different here, while the loader will report them as skipped....maybe `"\t".join` only the subset of `data` that is also used by loader to check for uniqueness?